tests/modules/misc/debug.nix

```diff
@@ -0,0 +1,19 @@
+{
+  debug = { pkgs, config, lib, ... }: {
+    home.enableDebugInfo = true;
+    home.packages = with pkgs; [ nix gdb ];
+
+    nmt.script = ''
+      [ -L $TESTED/home-path/lib/debug/nix ] \
+        || fail "Debug-symbols for pkgs.nix should exist in \`/home-path/lib/debug'!"
+      source $TESTED/home-path/etc/profile.d/hm-session-vars.sh
+      [[ "$NIX_DEBUG_INFO_DIRS" =~ /lib/debug$ ]] \
+        || fail "Invalid NIX_DEBUG_INFO_DIRS!"
+
+      # We need to override NIX_DEBUG_INFO_DIRS here as $HOME evalutes to the home
+      # of the user who executes this testcase :/
+      echo quit | NIX_DEBUG_INFO_DIRS=$TESTED/home-path/lib/debug gdb ./home-path/bin/nix 2>&1 | grep 'No debugging symbols found' \
+        && fail 'Failed to read debug symbols from nix in gdb' || true
```

Comment:
This one doesn't break the test if the `gdb`-call cannot read debug-symbols for nix :/

_goes back to `bash(1)`_